Missing depends

istoph opened the pull request:
Some bug fixes as the installation tests in the pipline fail.

istoph committed:
update: debian compat

istoph committed:
update: depends

istoph committed:
update: timeout to 3 sec for zabbix default and tab to char

istoph committed:
update copyright

istoph committed:
update: tab to space

imolein commented on debian/postinst:
```diff
@@ -21,13 +21,13 @@ case "$1" in
   configure)
 
-	chmod +x /usr/lib/zabbix-check/bin/zabbix-check-spamassassin
+    chmod +x /usr/lib/zabbix-check/bin/zabbix-check-spamassassin
```
Make the script executable before the packaging and remove this line

imolein commented on debian/postinst:
```diff
@@ -21,13 +21,13 @@ case "$1" in
-	else
-		/etc/init.d/zabbix-agent restart || true
-	fi	
+    if [ -x "`which invoke-rc.d 2>/dev/null`" ]; then
```
* Replace `which` with `command -v`, since `which` is not POSIX standart and maintscripts are POSIX shell scripts.
* Also a condition for systemd is missing.

imolein commented on usr/lib/zabbix-check/bin/zabbix-check-spamassassin:
```diff
@@ -1,21 +1,22 @@
 #!/bin/bash
```
Replace the shebang with `#!/usr/bin/env bash` or with `#!/usr/bin/env sh` since the script don't use any bash feature as far as I can see.

imolein commented on usr/lib/zabbix-check/bin/zabbix-check-spamassassin:
```diff
@@ -1,21 +1,22 @@
-else
-	echo 0
+if [ "$(netstat -ltn | grep 783 | wc -l)" -gt "0" ]; then
+    /etc/init.d/spamassassin status >/dev/null 2>&1
```
Is this working in a systemd environment?

istoph replied:
yes, systemd returns a status code in ubuntu

imolein replied:
Ah the sysvinit script calls systemd, I see.

imolein commented on debian/copyright:
```diff
@@ -1,16 +1,24 @@
-This package is currently maintained by Christoph Hueffelmann <chr@istoph.de> 2016 
+This package is actually maintained by Christoph Hüffelmann <chr@istoph.de> 2016
```
Change the copyright format to match the format from Debian's packaging manual: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/

imolein reviewed (commented):
See my comments on the code

istoph committed:
update: license to copyright-format/1.0

istoph committed:
update: fix postinstall after discussion

https://github.com/zabbix-deb/zabbix-check-spamassassin/pull/1#discussion_r563569523

istoph committed:
zabbix-check-spamassassin (0.1~dev4)

imolein commented on debian/postinst:
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/usr/bin/env bash
```
Why bash? The maintscripts use no bash feature. But if you insist in using bash here, you should add bash as `Pre-Depends` in control.

imolein replied:
Ah, maybe you read [this comment](https://github.com/zabbix-deb/zabbix-check-spamassassin/pull/1/files#r563567123) and changed the shebang in the wrong file?

istoph replied:
I still don't think that such systems can exist. `apt-cache show bash` says: `Priority: required`

https://www.debian.org/doc/debian-policy/ch-archive.html#s-priorities


>required
>
>    Packages which are necessary for the proper functioning of the system (usually, this means that dpkg functionality depends on these packages). Removing a required package may cause your system to become totally broken and you may not even be able to use dpkg to put things back, so only do so if you know what you are doing.
>
>    Systems with only the required packages installed have at least enough functionality for the sysadmin to boot the system and install more software.

istoph committed:
change postinst to sh

After a discussion, I agree with "sh", even if I am not actually convinced of it.

imolein committed:
changed shebang

imolein committed:
corrected copyright date

imolein reviewed (approved):
lgtm
